### ISSUE TITLE
Remove ObjectMapper

### ISSUE THREAD

**Issue by glennrfisher** [watson-developer-cloud/ios-sdk#181]
Let's investigate removing the ObjectMapper dependency by adding serialization/deserialization to each object. This provides the following benefits:

- Reduces the number of external dependencies (and hence Carthage build time, unexpected errors, and project size)
- Resolves the requirements from ObjectMapper for models to have `var` properties, even if we never change them.
- Resolves the requirements from ObjectMapper for models to be optional, even if we know they are non-optional in the JSON object.

Serialization is surprisingly elegant. Here is some sample code to see it in action:

```swift
protocol WatsonRequestModel {
    func toDictionary() -> [String: AnyObject]
}

extension WatsonRequestModel {

    func toJSONData(failure: (NSError -> Void)?) -> NSData? {
        let map = self.toDictionary()
        let data = try? NSJSONSerialization.dataWithJSONObject(map, options: [])

        guard let json = data else {
            if let failure = failure {
                let description = "Could not serialize \(self.dynamicType)"
                let domain = "swift.\(self.dynamicType)"
                let error = createError(domain, description: description)
                failure(error)
            }
            return nil
        }

        return json
    }

    func toJSONString(failure: (NSError -> Void)?) -> String? {
        guard let json = self.toJSONData(failure) else {
            return nil
        }

        let text = String(data: json, encoding: NSUTF8StringEncoding)
        return text
    }
}
```

```swift
func toDictionary() -> [String: AnyObject] {
    var map = [String: AnyObject]()
    map["action"] = action
    map["content-type"] = contentTypeString
    map["continuous"] = continuous
    map["max_alternatives"] = maxAlternatives
    map["interim_results"] = interimResults
    map["word_confidence"] = wordConfidence
    map["timestamps"] = timestamps
    map["keywords"] = keywords
    map["keywords_threshold"] = keywordsThreshold
    map["word_alternatives_threshold"] = wordAlternativesThreshold
    map["inactivity_timeout"] = inactivityTimeout
    return map
}
```

Here's a (very rough) example of how to deserialize JSON.

```swift
init?(jsonString: String) {
    guard let map = try? NSJSONSerialization.JSONObjectWithData(JSONData) else {
        return nil
    }

    guard let action = map["action"] as? String,
              let continuous = map["continuous"] as? Bool else
    {
        return nil
    }

    self.action = action
    self.continuous = continuous

    if let keywords = map["keywords"] as? String {
        self.keywords = keywords
    }
}
```

**Comment by glennrfisher** [comment]
This change will probably require Swift 2.2+. There is a bug in other versions of Swift that prevents returning `nil` from a failable initializer until all properties have been set to a value. That bug is fixed starting in Swift 2.2.

**Comment by glennrfisher** [comment]
We've addressed this problem in the `rest-kit` branch by using Freddy instead of ObjectMapper. Freddy is faster, safer, and provides a better architecture for us moving forwards.
